File: cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float change;
    int  total;
    do
    {
        change = get_float ("Amount owed - ");
        total = round(change*100);

    }while (change <= 0);

    int amount = 0;
    while(total >= 25) {
        amount++;
        total = total - 25;
    }


    while(total >= 10) {
        amount++;
        total = total - 10;
    }

    while(total >= 5) {
        amount++;
        total = total - 5;
    }

    while(total >= 1) {
        amount++;
        total = total - 1;
    }

    printf("%i\n", amount);
}

File: cash.py
from cs50 import get_float
from math import floor


def main():
    while True:
        change = get_float("Amount owed - ")
        total = floor(change * 100)

        if total > 0:
            break

    Q = total // 25
    D = (total%25) // 10
    N = ((total%25) % 10) // 5
    P = ((total%25) % 10) % 5

    print(f"{Q + D + N + P}")


if __name__ == "__main__":
    main()